docs/rpc_aio_leapfrog.md

```diff
@@ -0,0 +1,99 @@
+# RPC-AIO Leapfrog Gate Jobs
+
+Leapfrog periodic jobs are used to determine the health of an upgrade from a
+specific revision of a deployed version of RPC-Openstack to the latest supported
+revision in production.  They are in place to periodically test and will
+eventually gate on pull requests going into the RPC-O repositories until the
+leapfrogs have completed for each deployed version.
+
+The current configuration that generates the various leapfrog gate jobs are
+stored here:
+
+[https://github.com/rcbops/rpc-gating/blob/master/rpc_jobs/rpc_aio_leapfrog.yml
+](https://github.com/rcbops/rpc-gating/blob/master/rpc_jobs/rpc_aio_leapfrog.yml)
+
+The base template for AIO jobs resides in
+[rpc_aio.yml
+](https://github.com/rcbops/rpc-gating/blob/master/rpc_jobs/rpc_aio.yml#L186)
+so that we don't have to manage multiple templates.
+
+Individual jobs are generated from the configuration file in the following
+format which is what you'll see when you log into the Jenkins Server:
+
+    RPC-AIO_{series}-{image}-{action}-{scenario}-{ztrigger}
+
+So one example job might be:
+
+    RPC-AIO_kilo-r11.1.18-trusty-leapfrogupgrade-swift
+
+The configuration is broken up into several sections:
+
+##### Series
+
+The name of the job, the branch to test from rpc-openstack, and branches to run
+PR jobs on if present.
+
+#### Image
+
+The short form of the image name and image name as reflected in the Public Cloud
+Release Engineering account.  This is the image used as a base to run the tests
+on.
+
+#### Action
+
+This specifies the various actions and tasks that will be ran during the build.
```

Comment:
The action specifies what the test is doing -- ie. standard deploy, major upgrade, leapfrog upgrade, etc.  Since etc. action is responsible for different things we then override `ACTION_STAGES` with the necessary pipeline stages for that specific action.